cmd/probe/tcp_probe_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+)
+
+func TestTCPProbe(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), false)
+
+		if err := p.Run(t.Context()); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("conn should fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionSucceeded) {
+			t.Fatalf("expecting error %v, got %v", errConnectionSucceeded, err)
+		}
+	})
+
+	t.Run("conn should not fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		l.Close()
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionFailed) {
```

Comment:
```suggestion
		if err := p.Run(t.Context()); err != nil {
```

Reply:
I'm dismissing this because we want to confirm that the returned error is, indeed, `errConnectionFailed` (or any error that wraps it for that matter) and not something else (say a `context.Canceled` error).